packages/app/pages/submission/new.tsx

```diff
@@ -112,6 +114,8 @@ const NewSubmission = () => {
       setError("Failed to submit post.");
     },
     onSuccess() {
+      // clear posts on successful post
+      queryClient.invalidateQueries("posts");
```

Comment:
when a new post is added, invalidate cache of things that query with posts leading the query key.